test/models/detector/suggested_resource_test.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: detector_suggested_resources
+#
+#  id          :integer          not null, primary key
+#  title       :string
+#  url         :string
+#  phrase      :string
+#  fingerprint :string
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+require 'test_helper'
+
+module Detector
+  class SuggestedResourceTest < ActiveSupport::TestCase
+    test 'fingerprints are generated automatically' do
+      resource = {
+        title: 'Our latest resource',
+        url: 'https://example.org',
+        phrase: 'Our latest resource'
+      }
+
+      new_resource = Detector::SuggestedResource.create(resource)
+
+      assert new_resource.fingerprint == 'latest our resource'
+    end
+
+    test 'fingerprints are recalculated on save' do
+      resource = detector_suggested_resources('jstor')
+      refute resource.fingerprint == 'A brand new phrase'
+
+      resource.phrase = 'This is a brand new phrase'
+      resource.save
+      resource.reload
+
+      assert resource.fingerprint == 'a brand is new phrase this'
+    end
+
+    test 'fingerprints standardize characters used' do
```

Comment:
This test is covering a lot of functionality. It might be better to split it up into tests for each aspect of the fingerprint algorithm. I think some functionality may not have been included in the testing because it all got put in one. Notably missing (I think?) is de-duplication of the tokens. This test could be updated to include that, or we could have one test per "thing" the algorithm does.

Reply:
Yeah, I was tempted to split the tests into different features. I'm open to going that route, to be more explicit about what the different components are.